docs/development/go/go-structures/index.md

```diff
@@ -18,235 +19,532 @@ external_resources:
+}
+
+func main() {
+    employee1 := Employee{"Heather", 1910234}
+    employee2 := Employee{"Heather", 1910234}
+    fmt.Println(employee1 == employee2)
+}
+{{< /file >}}
+
+The output of `employee.go` will be:
+
+    go run employee.go
 
-    go run structures.go
 {{< output >}}
-0 0
-{23 Hello! -2}
-{0 Hi! 12}
-p: [{0 Hi! 12} {0  0} {23 Hello! -2}]
-p2: {1 Two 3.1}
-p: [{0 Hi! 12} {0  0} {23 Hello! -2}]
+true
 {{< /output >}}
 
 {{< note >}}
-Structures, in particular, and Go types, in general, are usually defined outside the `main()` function in order to have a global scope and be available to the entire Go package, unless you want to clarify that a type is only useful within the current scope and is not expected to be used elsewhere in your code.
+Structs cannot be ordered with operators like greater-than  `>` or less-than `<`.
 {{< /note >}}
 
-## Pointers and Structures
+### Accessing Fields
 
-As Go supports pointers, you can create pointers to structures. The use of pointer structures is illustrated in `pointers.go`.
+You can access a specific field using the struct variable name followed by a `.` character followed by the name of the field (also referred to as *dot notation*). Given an `Employee` variable named `mihalis`, the struct's two fields can be individually accessed as `mihalis.FirstName` and `mihalis.employeeID`:
 
-{{< file "./pointers.go" go >}}
+{{< file "employee.go" go >}}
 package main
 
 import (
     "fmt"
 )
 
-type myStructure struct {
-    Name    string
-    Surname string
-    Height  int32
+type Employee struct {
+    FirstName string
+    employeeID int
 }
 
 func main() {
-    ptr := &myStructure{"Mihalis", "Pointer", 123}
-    s2 := myStructure{"Mihalis", "s2", 123}
-
-    fmt.Println("Getting a specific structure field:", (*ptr).Surname)
-    fmt.Println(s2.Name, s2.Surname)
-    fmt.Println("As a pointer:", ptr)
-    fmt.Println("Memory address:", &ptr)
-    fmt.Println("Value:", *ptr)
-    fmt.Println("Value:", s2)
+    mihalis := Employee{"Mihalis", 1910234}
+    fmt.Println("My name is", mihalis.FirstName, "and my employee ID is", mihalis.employeeID)
 }
 {{< /file >}}
 
-`ptr` is a pointer variable whereas `s2` is a regular variable, which means that you will need the `*ptr` notation in order to dereference `ptr`. Additionally, if you want to get the value of a specific field of a pointer to a structure, you will need to use the `(*ptr).Surname` notation to separate the pointer from the field name. Last, notice that both `ptr` and `s2` are initialized in the code.
+The output of `employee.go` will be:
 
-The output of `pointers.go` will resemble the following:
+    go run employee.go
 
-    go run pointers.go
 {{< output >}}
-Getting a specific structure field: Pointer
-Mihalis s2
-As a pointer: &{Mihalis Pointer 123}
-Memory address: 0xc00009e000
-Value: {Mihalis Pointer 123}
-Value: {Mihalis s2 123}
+My name is Mihalis and my employee ID is 1910234
 {{< /output >}}
 
-## Functions and Structures
```

Comment:
I removed the stand-alone Functions and Structures section and split its contents into two different places:
- The Functions-with-structure-arguments example was integrated into the `Value Semantics with Functions` and the `Pointers and Structs and Functions` sections.
- The Functions-returning-structures example was integrated with the new Constructor Functions section further down

In place of a Functions and Structures section, I inserted a `Methods` section, which didn't seem to be mentioned in this guide, but did seem important when talking about structs. I also could not find this same content in one of the other open Go PRs. The Go Interfaces guide does talk about methods, but not quite in the same way. Because methods can be used with structs without using interfaces, so I wanted to introduce them earlier.